README.md

```diff
@@ -177,6 +177,31 @@ Call a gRPC endpoint using a JSON input. What this does behind the scenes:
 
 *See [docs/grpc.md](docs/grpc.md) for full instructions.*
 
+### SSL/TLS Connections
+
+To enable SSL/TLS connections to the server, use the `--tls` command line flag. 
```

Comment:
There's no equivalent mechanism in any other command, and this doesn't actually enable or disable the other flags as you say in the PR description - they are all installed regardless. I'm not sure if this is needed.

If we need this after discussion, then instead of ignoring fields below, they should error if set. So for example `--server-name` should result in an error if `--tls` is not set.

Reply:
We need the option to exist because of the simple case (i.e. TLS connection with default CA's trusted) so I like the option of an error if --tls is not set yet one of the other 4 parameters are specified.

Reply:
How does grpcurl handle this? It doesn't have a `--tls` flag equivalent.

Reply:
It assumes it by default. You have to pass a -plaintext flag to it for it to behave the way prototool grpc does now.  We could make prototool grpc function that way but it would change the default behavior of the tool.

Reply:
Got it. OK, let's go with the `--tls` flag and then error if TLS-specific flags are set without the TLS flag set (and document that they fail without the TLS flag)

Reply:
The primary idiosyncrasy between `grpcurl` and `curl` is that curl requires the transport protocol (i.e. https, http etc) where grpcurl does not.  Because of that, it has to be instructed to use a secure connection or not via some other flag.  By default grpcurl assumes a TLS connection and needs the -plaintext flag to function with an unencrypted connection.

Reply:
I think there is kind of a fundamental decision here. Do you want "prototool grpc" to function analogously with "curl" and therefore you should probably name all the flags the same (as grpcurl does) or do you want to create your own abstraction and only rely on grpcurl as the current implementation.

Reply:
Note that right now, you can actually set an address as either `tcp://address` or `unix://address` (not well documented) - we could expand this to accept `http` or `https` and use that as the mechanism, but honestly I think I prefer the `--tls` flag having thought about this a bit more.

Reply:
`prototool grpc` isn't functionally equivalent to either, it was just a last-minute addition before we released `prototool` - I'm actually working to document how to use other grpc-specific CLI tools with prototool for cases where grpc reflection isn't available (but this is a big TODO, and I want to add the ability to generate "mega-FileDescriptorSets" first).

That prototool uses grpcurl is just an implementation detail, and if we wanted, we might rip out grpcurl in the future if something else came along.

Reply:
How about this for a validation check:
```
	if tls {
		if (cert != "") != (key != "") {
			return newExitErrorf(255, "if cert is specified, key must be specified")
		} else if insecure && (cacert != "" || cert != "" || key != "" || serverName != "") {
			return newExitErrorf(255, "if insecure then cacert, cert, key, and server-name must not be specified")
		}
	} else if insecure || cacert != "" || cert != "" || key != "" || serverName != "" {
		return newExitErrorf(255, "tls must be specified if insecure, cacert, cert, key or server-name are specified")
	}
```

Reply:
or should we include the "--" with each flag in the errors:
```
	if tls {
		if (cert != "") != (key != "") {
			return newExitErrorf(255, "if --cert is specified, --key must be specified")
		} else if insecure && (cacert != "" || cert != "" || key != "" || serverName != "") {
			return newExitErrorf(255, "if --insecure then --cacert, --cert, --key, and --server-name must not be specified")
		}
	} else if insecure || cacert != "" || cert != "" || key != "" || serverName != "" {
		return newExitErrorf(255, "--tls must be specified if --insecure, --cacert, --cert, --key or --server-name are specified")
	}
```

Reply:
The only other way to support this without a flag would be to infer it based upon the prefix of the address. Right now we infer "tcp://" if the address doesn't have "tcp://" or "unix://" at the beginning. We could also allow one or all of "https" or "tls" or "ssl" or "ssh" to imply "tcp" with "tls".

But then the question becomes, should we also infer it if any of the other 4 flags are passed if no prefix is given in the address. What happens if the prefix is "tcp"?

Reply:
Yea let's just do the flag and not touch this part.